core/embed/rust/src/ui/model_mercury/component/keyboard/passphrase.rs

```diff
@@ -19,64 +20,134 @@ use crate::{
 };
 
 use core::cell::Cell;
+use num_traits::ToPrimitive;
 
 pub enum PassphraseKeyboardMsg {
     Confirmed,
     Cancelled,
 }
 
+/// Enum keeping track of which keyboard is shown and which comes next. Keep the
+/// number of values and the constant PAGE_COUNT in synch.
+#[repr(u32)]
+#[derive(Copy, Clone, ToPrimitive)]
+enum KeyboardLayout {
+    LettersLower = 0,
+    LettersUpper = 1,
+    Numeric = 2,
+    Special = 3,
+}
+
+impl KeyboardLayout {
+    fn next(self) -> Self {
+        match self {
+            Self::LettersLower => Self::LettersUpper,
+            Self::LettersUpper => Self::Numeric,
+            Self::Numeric => Self::Special,
+            Self::Special => Self::LettersLower,
+        }
+    }
+
+    fn prev(self) -> Self {
+        match self {
+            Self::LettersLower => Self::Special,
+            Self::LettersUpper => Self::LettersLower,
+            Self::Numeric => Self::LettersUpper,
+            Self::Special => Self::Numeric,
+        }
+    }
+}
+
+impl From<KeyboardLayout> for ButtonContent {
+    /// Used to get content for the "next keyboard" button
+    fn from(kl: KeyboardLayout) -> Self {
+        match kl {
+            KeyboardLayout::LettersLower => ButtonContent::Text("abc".into()),
+            KeyboardLayout::LettersUpper => ButtonContent::Text("ABC".into()),
+            KeyboardLayout::Numeric => ButtonContent::Text("123".into()),
+            KeyboardLayout::Special => ButtonContent::Icon(theme::ICON_ASTERISK),
+        }
+    }
+}
+
 pub struct PassphraseKeyboard {
     page_swipe: Swipe,
     input: Child<Input>,
-    back: Child<Button>,
-    confirm: Child<Button>,
+    input_prompt: Child<Label<'static>>,
+    erase_btn: Child<Maybe<Button>>,
+    cancel_btn: Child<Maybe<Button>>,
+    confirm_btn: Child<Maybe<Button>>,
+    next_btn: Child<Button>,
```

Comment:
One day we should drop `Child` from mercury entirely, as well as `paint_overlapping`, OK to keep it for now though.

Reply:
Yep, was thinking about that. 

Also here is an image depicting the futuristic scene of a `Child` being helped out of mercury
![image](https://github.com/trezor/trezor-firmware/assets/14001709/7cdc5dd3-38fc-403c-ab07-8fab24bfaad9)